boards/esp32-wroom-32/Makefile.dep

```diff
@@ -0,0 +1,49 @@
+include $(RIOTBOARD)/common/esp32/Makefile.dep
+
+# default parameter definitions when module enc28j60 is used
+ifneq (, $(filter enc28j60, $(USEMODULE)))
+
+    # avoid multiple definitions when package depenedencies are resolved recursively
+    ifndef ENC28J60_PARAM_DEFINED
+        export ENC28J60_PARAM_DEFINED = 1
+
+        # default definitions
+        ENC28J60_PARAM_SPI   ?= SPI_DEV\(0\)
+        ENC28J60_PARAM_CS    ?= GPIO32
+        ENC28J60_PARAM_INT   ?= GPIO35
+        ENC28J60_PARAM_RESET ?= GPIO33
+        CFLAGS += -DENC28J60_PARAM_SPI=$(ENC28J60_PARAM_SPI)
```

Comment:
This is suffering that currently `Makefile.dep` is not processed before `Makefile.include`. 

The `CFLAGS` should normally be added in the board `Makefile.include` and not here, but can't because dependencies are not resolved yet. There would then be no issues with "multiple definitions". 

I am trying to fix this globally, currently I only have the issue https://github.com/RIOT-OS/RIOT/issues/9913 and local wip PRs, but I am actively working on it.

I would prefer to have it fixed before introducing these.
But I would not block, and if the board is ready before, merge it and I will fix them after.
I am just saying I noticed it and it gives me even more reasons to fix the processing order.

Reply:
@cladmi Ups, I just forgot to remove these definitions from this ```Makefile.dep```. I will remove it.

I really had a dependency problem in ```cpu/esp32/Makefile.include```. There, I had to place some module dependency definitions since they are needed before the remaining of ```Makefile.include``` is parsed and ```Makefile.dep``` is included.

Reply:
You may not be able to change it for the moment because of the dependency resolution order. If you can't just keep it there no worry.

It is a bit more problematic for the module dependencies, I will try to move forward with fixing the order as soon as possible so it can be fixed.

Reply:
I will remove thm anyway, since it was not my intention to define them at all. I just forgot it for this board :worried:

Reply:
Ok no problem :)